Alignment/OfflineValidation/src/PrepareDMRTrends.cc

```diff
@@ -0,0 +1,160 @@
+    f->Close();
+  }
+
+  TFile *fout = TFile::Open(outputFileName_, "RECREATE");
+  for (TString &structure : structures) {
+    TString structname = structure;
+    structname.ReplaceAll("_y", "");
+    size_t layersnumber = nlayers.at(structname);
+    for (size_t layer = 0; layer <= layersnumber; layer++) {
+      for (const string &geometry : geometries) {
+        TString name = Variable + "_" + getName(structure, layer, geometry);
+        Geometry geom = mappoints[make_pair(make_pair(structure, layer), geometry)];
+        using Trend = vector<float> (Geometry::*)() const;
+        vector<Trend> trends{&Geometry::Mu,
+                             &Geometry::Sigma,
+                             &Geometry::MuPlus,
+                             &Geometry::SigmaPlus,
+                             &Geometry::MuMinus,
+                             &Geometry::SigmaMinus,
+                             &Geometry::DeltaMu,
+                             &Geometry::SigmaDeltaMu};
+        vector<TString> variables{
+            "mu", "sigma", "muplus", "sigmaplus", "muminus", "sigmaminus", "deltamu", "sigmadeltamu"};
+        vector<float> runs = geom.Run();
+        size_t n = runs.size();
+        vector<float> emptyvec;
+        for (size_t i = 0; i < runs.size(); i++)
+          emptyvec.push_back(0.);
+        for (size_t iVar = 0; iVar < variables.size(); iVar++) {
+          Trend trend = trends.at(iVar);
+          TGraphErrors *g = new TGraphErrors(n, runs.data(), (geom.*trend)().data(), emptyvec.data(), emptyvec.data());
+          g->SetTitle(geometry.c_str());
+          g->Write(name + "_" + variables.at(iVar));
+        }
+        vector<pair<Trend, Trend>> trendspair{make_pair(&Geometry::Mu, &Geometry::Sigma),
+                                              make_pair(&Geometry::MuPlus, &Geometry::SigmaPlus),
+                                              make_pair(&Geometry::MuMinus, &Geometry::SigmaMinus),
+                                              make_pair(&Geometry::DeltaMu, &Geometry::SigmaDeltaMu)};
+        vector<pair<TString, TString>> variablepairs{make_pair("mu", "sigma"),
+                                                     make_pair("muplus", "sigmaplus"),
+                                                     make_pair("muminus", "sigmaminus"),
+                                                     make_pair("deltamu", "sigmadeltamu")};
+        for (size_t iVar = 0; iVar < variablepairs.size(); iVar++) {
+          Trend meantrend = trendspair.at(iVar).first;
+          Trend sigmatrend = trendspair.at(iVar).second;
+          TGraphErrors *g = new TGraphErrors(
+              n, runs.data(), (geom.*meantrend)().data(), emptyvec.data(), (geom.*sigmatrend)().data());
+          g->SetTitle(geometry.c_str());
+          TString graphname = name + "_" + variablepairs.at(iVar).first;
+          graphname += variablepairs.at(iVar).second;
+          g->Write(graphname);
+        }
+      }
+    }
+  }
+  fout->Close();
+}
```

Comment:
the things that are `new`-ed technically should be `delete`-ed as well in the end

Reply:
Pointers properly dealt with.